File: OCP/Ruby/não aplicado/carrinho.rb
class Carrinho
    def valorTotal
        total = 0.0
        items.each do | item |
            if item.medida == "unidade"
                total += item.quantidade * 5
            elsif item.medida == "peso"
                total += item.quantidade * 4.0 / 1000 # quantidade em gramas, preço por kg
            elsif item.medida == "especial"
                total += item.quantidade * 0.4
                pacoteDeTres = item.quantidade / 3
                total -= pacoteDeTres * 0.2
            end
            # mais regras podem vir
        end
    end
end

File: OCP/C#/não aplicado/Carrinho.cs
class Carrinho
{
    public List<ItemCarrinho> Itens { get; set; }
    
    public decimal ValorTotal 
    {
        get
        {
            var soma = 0;
            foreach (var item in Itens)
            {
                if (item.Medida == "unidade")
                    soma += item.Quantidade * 5;
                else if (item.Medida == "peso")
                    soma += item.Quantidade * 4 / 1000;
                else if (item.Medida == "especial")
                {
                    var valor        = item.Quantidade * 0.4;
                    var pacoteDeTres = item.Quantidade / 3;
                    soma += valor - pacoteDeTres * 0.2;
                }
                // mais regras podem vir...esse código não vai parar de crescer...
            }
            return soma;
        }
    }
}